app/src/androidTest/java/com/example/android/justjava/Models/JustJavaModel.kt

```diff
@@ -0,0 +1,214 @@
+    fun verifyMinCount() {
+        closeSoftKeyboard()
+        clickMinusButton()
+        clickMinusButton()
+        clickMinusButton()
+        clickMinusButton()
+        onView(withText(R.string.min_count_toast)).inRoot(ToastMatcher()).check(matches(withText(YOU_CANNOT_HAVE_LESS_THAN_1_COFFEE)))
+        quantityDisplay.check(matches(withText("1")))
+    }
+
+
+    // click plus button
+    private fun clickPlusButton() {
+        plusButton.perform(click())
+    }
+
+    // check that count has increased after clicking plus button
+    fun verifyIncreasedCount() {
+        clickPlusButton()
+        quantityDisplay.check(matches(withText("2")))
+    }
+
+    // check that count does not exceed 5, verify toast message
+    fun verifyMaxCount() {
+        closeSoftKeyboard()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        onView(withText(R.string.max_count_toast)).inRoot(ToastMatcher()).check(matches(withText(MAX_5_COFFEES_PER_ORDER)))
+        quantityDisplay.check(matches(withText("5")))
+    }
+
+    // verify that all expected UI elements are on the screen
+    fun verifyAllDefaultElements() {
+
+        closeSoftKeyboard()
+        nameEditText.check(matches(isDisplayed()))
+        toppingsHeader.check(matches(isDisplayed()))
+        creamCheckBox.check(matches(isDisplayed()))
+        chocolateCheckBox.check(matches(isDisplayed()))
+        quantityHeader.check(matches(isDisplayed()))
+        minusButton.check(matches(isDisplayed()))
+        quantityDisplay.check(matches(isDisplayed()))
+        plusButton.check(matches(isDisplayed()))
+        summaryHeader.check(matches(isDisplayed()))
+        orderButton.check(matches(isDisplayed()))
+        totalAmount.check(matches(isDisplayed()))
+    }
+
+    // clicks the order button
+    private fun clickOrderButton() {
+        orderButton.perform(click())
+    }
+
+
+    // add name, check chocolate box, increase quantity to 2, submit order
+    fun submitOrder1() {
+        addNameToEditText(firstNameOnly)
+        closeSoftKeyboard()
+        clickChocolateCheckbox()
+        clickPlusButton()          // how to verify intent and its contents?
+        clickOrderButton()
+        //testEmailIntent();
+        totalAmount.check(matches(withText(order1)))
+        totalAmount.check(matches(withText(containsString(`TOTAL_$14`))))
+
+    }
+
+    // add long name, check chocolate box, check cream box,  increase quantity to 6, decrease to 4, submit order
+    fun submitOrder2() {
+        exceedCharLimit()
+        closeSoftKeyboard()
+        clickWhippedCreamCheckbox()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        clickPlusButton()
+        onView(withText(R.string.max_count_toast)).inRoot(ToastMatcher()).check(matches(withText(MAX_5_COFFEES_PER_ORDER)))
```

Comment:
"MAX_5_COFFEES_PER_ORDER" is coming from the companion object you created below. Is this the same as R.string.max_count_toast? I think you could just verify the resource string of R.string.max_count_toast is displayed. That would look like this... onView(withText(R.string.max_count_toast)).inRoot(ToastMatcher()).check(matches(isDisplayed))